src/java/htsjdk/samtools/SAMRecordSetBuilder.java

```diff
@@ -346,6 +347,7 @@ public void addPair(final String name, final int contig, final int start1, final
         end2.setAlignmentStart(start2);
         end2.setReadNegativeStrandFlag(true);
         end2.setCigarString(readLength + "M");
+        end2.setAttribute(ReservedTagConstants.NM,0);
```

Comment:
I don't think this is the only place where ```NM``` is not set (see the other ```addPair``` methods).  Do you want to add it to ```createReadNoFlag```?